docs/config-v3-documentation.md

```diff
@@ -0,0 +1,953 @@
+- list[T] — Space separated list of values of type T.
+
+## Reserved values
+Values beginning with `!` or `@` are reserved and have special meaning:
+
+- Only `!` value is `!unset`, which makes the key have no value (and thus ignore [higher level defaults](#use))
+- `@` values autoexpand depending on context
+
+
+---
+
+## [task]
+
+### version
+<!-- md:version 1.0.0 -->
+<!-- md:type enum -->
+<!-- md:default v1 -->
+
+Version of the task configuration file:
+
+- `v1` - Old version with a basic set of options.
+- `v2` - Old version, with an expanded set of options.
+- `v3` - Current version, **recommended**
+
+The version applies to this config file only and cannot be inherited with the use key.
+You can use `pisek config update` to update the config to the highest version.
+
+### use
+<!-- md:version 1.1.0 -->
+<!-- md:type path -->
+<!-- md:default-empty -->
+
+Config to use defaults from
+
+Values of keys are loaded in the following way:
+
+1. We try to find a given (section, key) in the task config
+2. If not found, we go to the config specified by the use key.
+3. If the use key is empty, we go to pisek's global defaults.
+4. If we can't find the (section, key) there, we reset to the task config and
+   try searching for the defaulting section and key of this key.
+5. If there is no defaulting key, we fail as this key is required.
+
+For example, consider:
+
+- `use=organization-config`
+- (`[section]`,`key`) defaults to (`[default_section]`,`key`)
+- (`[default_section]`,`key`) defaults to `42` if not present
+
+Then we search in this order:
+
+| config                  | section             | key   |
+|-------------------------|---------------------|-------|
+| this config             | `[section]`         | `key` |
+| organization-config     | `[section]`         | `key` |
+| pisek's global-defaults | `[section]`         | `key` |
+| this config             | `[default_section]` | `key` |
+| organization-config     | `[default_section]` | `key` |
+| pisek's global-defaults | `[default_section]` | `key` | 
+
+### task_type
+<!-- md:version 1.0.0 -->
+<!-- md:type enum -->
+<!-- md:default batch -->
+
+Task type:
+
+- `batch`
+- `interactive`
+
+### score_precision
+<!-- md:version 1.4.0 -->
+<!-- md:type int -->
+<!-- md:default 0 -->
+
+How many decimal digits scores are rounded to.
+
+## [tests]
+
+### in_gen
+<!-- md:version 1.0.0 -->
+<!-- md:type run_ref -->
+<!-- md:flag required -->
+
+Reference to the run section specifying how to run the generator. (If empty, only static inputs are used.)
+
+See [`[run]`](#run) for more.
+
+### gen_type
+<!-- md:version 1.3.0 -->
+<!-- md:type enum -->
+<!-- md:flag required-applicable -->
+
+Specifies the generator type. (Required for non-empty [`in_gen`](#in_gen).)
+
+- `pisek-v1` (recommended)
+- `cms-old`
+- `opendata-v1`
+
+For more see [generator](./task-parts/generator.md).
+
+### in_format
+<!-- md:version 1.7.0 -->
+<!-- md:type enum -->
+<!-- md:default strict-text -->
+
```

Comment:
It might be better to at least refer to another key's description instead of leaving a description empty.